content/docs/guides/assertions.md

```diff
@@ -11,27 +11,115 @@ An assertion query is written to find rows that violate one or more rules. If th
 
 ![Assertions](/static/images/assertions.png)
 
-## Example: Null checks
+## Auto generated assertions
+
+<div className="bp3-callout bp3-icon-info-sign" markdown="1">
+  Auto generated assertions are only supported from <code>@dataform/core</code> version <code>1.4.13</code>.
+</div>
+
+Dataform provides a convenient way to define assertions for tables, views and incremental tables as part of the `config` block of the table's SQLX file.
+
+Here's a complete example:
+
+`definitions/my_table.sqlx:`
+
+```js
+config {
+  type: "table",
+  assertions: {
+    uniqueKey: ["user_id"],
+    nonNull: ["user_id", "customer_id"],
+    rowConditions: [
+      'signup_date is null or signup_date > "2019-01-01"',
+      'email like "%@%.%"'
+    ]
+  }
+}
+select ...
+```
+
+For complete and up to date configuration options, view the [reference documentation for table assertions](/reference#ITableAssertions).
+
+## Unique keys
+
+If the `uniqueKey` property is set, the resulting assertion will fail if there is more than one row in the dataset with the same values for all of the column(s).
+
+```js
+config {
+  type: "table",
+  assertions: {
+    uniqueKey: ["user_id"]
+  }
+}
+select ...
+```
+
+Multiple key values can be provided.
+
+The generated assertion will be called: `<original_table_name>_assertions_uniqueKey`.
+
+## Non-null checks
+
+To quickly enforce that a set of columns are never null, provide an array of the column names with the `nonNull` property:
+
+```js
+config {
+  type: "table",
+  assertions: {
+    nonNull: ["user_id", "customer_id", "email"]
+  }
+}
+select ...
+```
+
+The generated assertion will be called: `<original_table_name>_assertions_rowConditions`, as this is just a shorthand for custom row conditions (see below).
```

Comment:
```suggestion
The generated assertion will be called `<original_table_name>_assertions_rowConditions`, because `nonNull` is just shorthand for custom row conditions (see below).
```